ktlint-ruleset-standard/src/main/kotlin/com/pinterest/ktlint/ruleset/standard/rules/ChainMethodContinuation.kt

```diff
@@ -0,0 +1,195 @@
+package com.pinterest.ktlint.ruleset.standard.rules
+
+import com.pinterest.ktlint.rule.engine.core.api.ElementType.BLOCK_COMMENT
+import com.pinterest.ktlint.rule.engine.core.api.ElementType.DOT
+import com.pinterest.ktlint.rule.engine.core.api.ElementType.EOL_COMMENT
+import com.pinterest.ktlint.rule.engine.core.api.ElementType.EXCLEXCL
+import com.pinterest.ktlint.rule.engine.core.api.ElementType.RBRACE
+import com.pinterest.ktlint.rule.engine.core.api.ElementType.SAFE_ACCESS
+import com.pinterest.ktlint.rule.engine.core.api.Rule
+import com.pinterest.ktlint.rule.engine.core.api.firstChildLeafOrSelf
+import com.pinterest.ktlint.rule.engine.core.api.hasNewLineInClosedRange
+import com.pinterest.ktlint.rule.engine.core.api.indent
+import com.pinterest.ktlint.rule.engine.core.api.isWhiteSpace
+import com.pinterest.ktlint.rule.engine.core.api.isWhiteSpaceWithNewline
+import com.pinterest.ktlint.rule.engine.core.api.lastChildLeafOrSelf
+import com.pinterest.ktlint.rule.engine.core.api.leavesIncludingSelf
+import com.pinterest.ktlint.rule.engine.core.api.nextCodeSibling
+import com.pinterest.ktlint.rule.engine.core.api.noNewLineInOpenRange
+import com.pinterest.ktlint.rule.engine.core.api.prevCodeSibling
+import com.pinterest.ktlint.rule.engine.core.api.prevLeaf
+import com.pinterest.ktlint.rule.engine.core.api.upsertWhitespaceBeforeMe
+import com.pinterest.ktlint.ruleset.standard.StandardRule
+import org.jetbrains.kotlin.com.intellij.lang.ASTNode
+import org.jetbrains.kotlin.com.intellij.psi.tree.TokenSet
+
+/**
+ * When present, align chain operators('.' or '?.') at previous line with next method call
+ */
+public class ChainMethodContinuation :
+    StandardRule("chain-method-continuation"),
+    Rule.Experimental {
+    private val chainOperators = TokenSet.create(DOT, SAFE_ACCESS)
+
+    override fun beforeVisitChildNodes(
+        node: ASTNode,
+        autoCorrect: Boolean,
+        emit: (offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit,
+    ) {
+        node
+            .takeIf { it.elementType in chainOperators }
+            ?.let { visitChainOperator(node, autoCorrect, emit) }
+    }
+
+    private fun visitChainOperator(
+        chainOperator: ASTNode,
+        autoCorrect: Boolean,
+        emit: (offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit,
+    ) {
+        val rightExpression = chainOperator.nextCodeSibling()?.firstChildLeafOrSelf() ?: return
+        val previousEndBrace =
+            chainOperator.getPrevChainEndBrace() ?: return
+        val previousExpressionBeforeBrace = previousEndBrace.prevCodeSibling()?.lastChildLeafOrSelf() ?: return
+        val isPreviousChainElementMultiline =
+            previousEndBrace.elementType == RBRACE &&
```

Comment:
hi @paul-dingemans shuld we also check `RBRACKET` and `RPAR` here?

Reply:
Might be, but a code sample that you would like to detect would be helpful.

But you can also look at it another way. I think that you want to detect that there is no newline between end of the last code leaf before the chain operator and the start of the first code leaf after the chain operator. So then it doesn't matter whether it is RBRACE, RBRACKET or RPAR.

Reply:
`I think that you want to detect that there is no newline between the end of the last code leaf before the chain operator and the start of the first code leaf after the chain operator.` yes for cases when RBRACE is in multiline. I got this idea when in a few instances in my PR you pointed out that the below code is not indented
```
listOf(1, 2, 3)
    .map {
        it*2
    }
    .sum()
```
and you suggested below alternative
```
listOf(1, 2, 3)
    .map {
        it*2
    }.sum()
```